src/main/java/com/y/java_board/controller/ArticleController.java

```diff
@@ -0,0 +1,104 @@
+package com.y.java_board.controller;
+
+import com.y.java_board.domain.Article;
+import com.y.java_board.domain.Comment;
+import com.y.java_board.dto.ArticleDto;
+import com.y.java_board.dto.CommentDto;
+import com.y.java_board.service.ArticleService;
+import com.y.java_board.service.CommentService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+@Controller()
+public class ArticleController {
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+    private final ArticleService articleService;
+    private final CommentService commentService;
+
+    public ArticleController(ArticleService articleService, CommentService commentService){
+        this.articleService = articleService;
+        this.commentService = commentService;
+    }
+
+    @GetMapping("/articles")
+    public String articles(Model model){
+        List<Article> articles = articleService.findArticles();
+        model.addAttribute("articles", articles);
+        return "article/articles";
+    }
+
+    @GetMapping("/articles/new")
+    public String showCreateForm(Model model){
+        model.addAttribute("articleDto", new ArticleDto("","",""));
+        return "article/create";
+    }
+
+    @PostMapping("/articles/new")
+    public String create(ArticleDto articleDto, BindingResult result, Model model){
+        if(result.hasErrors()){
+            // model.addAttribute() 오류의 원인을 보낸다.
+            return "/articles/new";
+        }
+        articleService.createOne(articleDto.toEntity());
+        return "redirect:/articles";
+    }
+
+    @GetMapping("/articles/{id}")
+    public String showArticle(@PathVariable("id") long id, Model model){
+        Optional<Article> articleOptional = articleService.findOne(id);
+        if(articleOptional.isPresent()){
+            List<Comment> comments = commentService.findCommentsByArticleId(id);
+            model.addAttribute("article",articleOptional.get());
+            model.addAttribute("commentDto", new CommentDto("", "", id));
+            model.addAttribute("comments", comments);
+            return "/article/detail";
+        }
+        // TODO : 해당 id 가 없다는 안내가 필요할 것 같다.
+        return "redirect:/articles";
+    }
+
+    @GetMapping("/articles/delete/{id}")
```

Comment:
Please do not have verbs in API. If you want to delete an article, go for DELETE request.

Reply:
while using the proper mapping method, the api name has also imporved! Now I know that api name should not use verb!